pkg/generators/models/models.go

```diff
@@ -189,13 +198,34 @@ func NewModelFromParameters(params openapi3.Parameters) (model *Model, err error
 	return model, nil
 }
 
-func (m *Model) fillConvertSpecs(ref *openapi3.SchemaRef) {
+func (m *Model) configureOneOf(ref *openapi3.SchemaRef) {
 	for _, oneOf := range ref.Value.OneOf {
 		m.ConvertSpecs = append(m.ConvertSpecs, ConvertSpec{
 			TargetGoType: goTypeFromSpec(oneOf),
 		})
 	}
 
+	if ref.Value.Discriminator != nil {
+		m.Discriminator.Field = ref.Value.Discriminator.PropertyName
+		m.Discriminator.Map = map[string]string{}
+
+		for value, ref := range ref.Value.Discriminator.Mapping {
+			m.Discriminator.Map[value] = typeNameFromRef(ref)
+		}
+
+		// the default mapping is to use the name of the model as the field value.
```

Comment:
this comes directly from the openapi docs https://swagger.io/docs/specification/data-models/inheritance-and-polymorphism/